### PR TITLE
Add getting start guidance for dllib

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

**Comment by jason-dai** [comment]
Make it more detailed "getting start tutorial" - describing the needed steps for each task, with links to the document website for more details. Need to add the following tasks to the document:

* Creating dev environment
   - Scala project (maven & sbt)
   - IDE (InteliJ)
* Code initialization
   - initNNContext – for local, YARN and "Job Server" (like Databricks and Zeppelin)
* Distributed data processing
   - Using Spark Dataframe APIs
* Model definition
   - Using Keras-like APIs
* Distributed model training
* Model saving and loading
* Distributed evaluation and inference
* Running program

Afterwards, we may add the following to the document:
* Checkpointing and resuming training
* Monitor the training
  - tensorboard
* Transfer learning and finetuning
  - freeze and trainable
* Hyperparameter tuning
  - optimizer
  - learning rate schedule
  - batch size
  - regularizer
  - clipping

File: docs/readthedocs/source/doc/DLlib/Overview/freeze.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+## Module Freeze
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/freeze.md:1]: Model Freeze

File: docs/readthedocs/source/doc/DLlib/Overview/freeze.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+To "freeze" a module means to exclude some layers of model from training.
+
+```scala
+module.freeze("layer1", "layer2")
+module.unFreeze("layer1", "layer2")
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/freeze.md:6]: model.freeze

File: docs/readthedocs/source/doc/DLlib/Overview/freeze.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+```
+* The whole module can be "freezed" by calling ```freeze()```. If a module is freezed,
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/freeze.md:8]: model instead of module

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
+# DLLib Getting Start Tutorial
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:1]: DLLib Getting Started Guide

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+</dependency>
+```
+
+- **SBT**
+```
+libraryDependencies += "com.intel.analytics.bigdl" % "bigdl-dllib-spark_2.4.6" % "0.14.0"
+```
+For more information about how to add BigDL dependency, please refer https://bigdl.readthedocs.io/en/latest/doc/UserGuide/scala.html#build-a-scala-project
+
+#### IDE (Intelij)
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:24]: See https://bigdl-project.github.io/master/#ScalaUserGuide/install-build-src/#setup-ide

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+Open up IntelliJ and click File => Open
+
+Navigate to your project. If you have add BigDL DLLib as dependency in your pom.xml.
+The IDE will automatically download it from maven and you are able to run your application.
+
+
+## 2. Code initialization
+```NNContext``` is the main entry for provisioning the dllib program on the underlying cluster (such as K8s or Hadoop cluster), or just on a single laptop.
+
+We usually do some initialization at first
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:34]: It is recommended to initialize `NNContext` at the beginning of your program:

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+## 3. Distributed Data Processing
+
+#### Using Spark Dataframe APIs
+DLlib supports Spark Dataframes as the input to the distributed training, and as
+the input/output of the distributed inference. Consequently, the user can easily
+process large-scale dataset using Apache Spark, and directly apply AI models on
+the distributed (and possibly in-memory) Dataframes without data conversion or serialization
+
+We used [Pima Indians onset of diabetes](https://raw.githubusercontent.com/jbrownlee/Datasets/master/pima-indians-diabetes.data.csv) as dataset for the demo. It's a standard machine learning dataset from the UCI Machine Learning repository. It describes patient medical record data for Pima Indians and whether they had an onset of diabetes within five years.
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:50]: Is this really a demo? I don't think you have a demo program in this guide.

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+Process the DataFrame to create the label and features. Then split it into traing part and validation part
+
+```
+val assembler = new VectorAssembler()
+  .setInputCols(Array("num_times_pregrant", "plasma_glucose", "blood_pressure", "skin_fold_thickness", "2-hour_insulin", "body_mass_index", "diabetes_pedigree_function", "age"))
+  .setOutputCol("features")
+val assembleredDF = assembler.transform(df)
+val df2 = assembleredDF.withColumn("label",col("class").cast(DoubleType) + lit(1))
+val Array(trainDF, valDF) = df2.randomSplit(Array(0.8, 0.2))
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:77]: are we using double or float?

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+Now we have got the data which is ready to train. Next we will build a deep learning model using DLLib Keras API
+
+## 4. Model Definition
+
+#### Using Keras-like APIs
+
+To define a model, you can use the [Keras Style API](https://bigdl.readthedocs.io/en/latest/doc/DLlib/Overview/keras-api.html).
+```
+import com.intel.analytics.bigdl.dllib.keras.layers._
+val x1 = Input[Float](Shape(8))
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:89]: The user needs not to use `[Float]` in the program; instead, I think the user can just `import com.intel.analytics.bigdl.numeric.NumericFloat`, and then the type can be inferred.

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+## 9. Checkpointing and resuming training
+You can configure periodically taking snapshots of the model.
+```
+dmodel.setCheckpoint(path, overWrite=False)
+```
+You can also set ```overWrite``` to ```True``` to enable overwriting any existing snapshot files
+
+After training stops, you can resume from any saved point. Choose one of the model snapshots to resume (saved in checkpoint path, details see Checkpointing). Use Models.loadModel to load the model snapshot into an model object.
+```
+val loadModel = Models.loadModel[Float](path)
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:192]: need to support type infer for `loadModel` as well.

File: docs/readthedocs/source/doc/DLlib/Overview/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+```
+dmodel.predict(df, featureCols = Array("features"), predictionCol = "predict")
+```
+
+- **evaluation**
+```
+dmodel.evaluate(df, featureCols = Array("features"))
+```
+
+## 8. Running program
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/DLlib/Overview/getting-started.md:144]: Move this to the end of the guide
